net: mqtt: Return -1 if keepalive messages are disabled.

simensrostad opened the pull request:
In mqtt_keepalive_time_left(), return -1 if keep alive messages are
disabled by setting CONFIG_MQTT_KEEPALIVE=0.

This allows to use mqtt_keepalive_time_left() directly as an input
for poll(). If no keep-alive is expected, -1 would indicate
that poll() can block until new data is available on the socket.

simensrostad committed:
net: mqtt: Return -1 if keepalive messages are disabled.

In mqtt_keepalive_time_left(), return -1 if keep alive messages are
disabled by setting CONFIG_MQTT_KEEPALIVE=0.

This allows to use mqtt_keepalive_time_left() directly as an input
for poll(). If no keep-alive is expected, -1 would indicate
that poll() can block until new data is available on the socket.

Signed-off-by: Simen S. Røstad <simen.rostad@nordicsemi.no>

rlubos reviewed (approved):
LGTM

jukkar commented:
The sanitycheck failure is unrelated to this PR, can you rebase against latest upstream and force push.

jukkar reviewed (commented):
As this is an API change, please add a description of the change to `doc/releases/release-notes-2.5.rst` file. There is a chapter `API Changes` where this should be documented.

simensrostad commented:
> As this is an API change, please add a description of the change to `doc/releases/release-notes-2.5.rst` file. There is a chapter `API Changes` where this should be documented.

Rebased against upstream and added description to release notes.
